src/Microsoft.SqlTools.ServiceLayer/Security/SecurityService.cs

```diff
@@ -317,7 +317,7 @@ await requestContext.SendResult(new LoginViewInfo()
                 SupportWindowsAuthentication = prototype.WindowsAuthSupported,
                 SupportAADAuthentication = prototype.AADAuthSupported,
                 SupportSQLAuthentication = true, // SQL Auth support for login, not necessarily mean SQL Auth support for CONNECT etc.
-                CanEditLockedOutState = true,
+                CanEditLockedOutState = !parameters.IsNewObject && prototype.IsLockedOut,
```

Comment:
looks better, but is this how SSMS determines the state?

Reply:
also, have you tested the unlocking?

Reply:
I've tested unlocking with latest iteration and it's working fine.